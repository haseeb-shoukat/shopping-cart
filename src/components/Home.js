import { useEffect } from "react";
import backgroundImage from "../images/bg.svg";

const Home = () => {
  useEffect(() => {
    document.body.style.backgroundImage = `url(${backgroundImage})`;
    document.querySelector(".active").classList.remove("active");
    document.querySelector("#home").classList.add("active");
  }, []);

  return (
    <div className="home">
      <div className="home-card">
        <div className="home-heading">BEST ONLINE STORE OF THE YEAR</div>
        <div className="home-quote">
          Style is a way to say who you are without having to speak
        </div>
      </div>
      <img
        className="home-img"
        src="./images/home-cover.jpg"
        alt="cover image"
      />
    </div>
  );
};

export default Home;
