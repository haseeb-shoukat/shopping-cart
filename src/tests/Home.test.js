import React from "react";
import { render } from "@testing-library/react";
import Home from "../components/Home";

describe("Home component", () => {
  it("Renders successfully", () => {
    const home = global.document.createElement("div");
    home.id = "home";
    global.document.body.appendChild(home);
    const { container } = render(<Home />);
    expect(container).toMatchSnapshot();
  });
});
