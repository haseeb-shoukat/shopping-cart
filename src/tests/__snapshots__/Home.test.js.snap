// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component Renders successfully 1`] = `
<div>
  <div
    class="home"
  >
    <div
      class="home-card"
    >
      <div
        class="home-heading"
      >
        BEST ONLINE STORE OF THE YEAR
      </div>
      <div
        class="home-quote"
      >
        Style is a way to say who you are without having to speak
      </div>
    </div>
    <img
      alt="cover image"
      class="home-img"
      src="./images/home-cover.jpg"
    />
  </div>
</div>
`;
